docs/config/node.rst

```diff
@@ -746,6 +746,33 @@ Queries
     ``ANY`` operators on arrays of that length can lead to heavy memory,
     consumption which could cause nodes to crash with OutOfMemory exceptions.
 
+Legacy
+=======
+
+.. _legacy.table_function_column_naming:
+
+**legacy.table_function_column_naming**
+  | *Default:* ``false``
+  | *Runtime:* ``no``
+
+  Since CrateDB 5.0.0, if the table function is not aliased and is returning a
+  single base data typed column, the table function name is used as the column
+  name. This setting can be set in order to use the naming convention prior to
+  5.0.0.
+
+  .. NOTE::
+
+    To be precise, the table functions affected are ``unnest``,
+    ``regexp_matches``, and ``generate_series``. When the setting is set and
+    a single column is expected to be returned, the returned column will be
+    named ``col1``, ``groups``, or ``col1`` respectively.
+    Any other table functions defined after CrateDB 5.0.0 will not be affected.
```

Comment:
I think this should be part of the normal documentation, I see no need for put it into a Note.
Also, if all affected functions are explicit listed, I see no need to mention that future function will not be affected.

```suggestion
  The following table functions are affected by this setting:

  - :ref:`unnest`,
  - :ref:`regexp_matches <table-functions-regexp-matches>`
  - :ref:`generate_series <table-functions-generate-series>`

  When the setting is set and a single column is expected to be returned,
  the returned column will be named ``col1``, ``groups``, or ``col1``
  respectively.
```